resources/Microsoft.VSCode.Dsc/Microsoft.VSCode.Dsc.psm1

```diff
@@ -5,44 +5,77 @@ $ErrorActionPreference = 'Stop'
-    $installLocationProperty = 'InstallLocation'
+    $architecture = [System.Runtime.InteropServices.RuntimeInformation]::OSArchitecture
 
-    if ($Insiders) {
-        $cmdPath = 'bin\code-insiders.cmd'
-        $insidersUserInstallLocation = TryGetRegistryValue -Key "$($userUninstallRegistry)\$($vsCodeInsidersUserProductCode)" -Property $installLocationProperty
-        if ($insidersUserInstallLocation) {
-            return $insidersUserInstallLocation + $cmdPath
-        }
+    switch ($architecture) {
+        'X64' { return $Insiders ? @('{217B4C08-948D-4276-BFBB-BEE930AE5A2C}_is1', '{1287CAD5-7C8D-410D-88B9-0D1EE4A83FF2}_is1') : @('{771FD6B0-FA20-440A-A002-3B3BAC16DC50}_is1', '{EA457B21-F73E-494C-ACAB-524FDE069978}_is1') }
+        'X86' { return $Insiders ? @('{C26E74D1-022E-4238-8B9D-1E7564A36CC9}_is1', '{26F4A15E-E392-4887-8C09-7BC55712FD5B}_is1') : @('{D628A17A-9713-46BF-8D57-E671B46A741E}_is1', '{F8A2A208-72B3-4D61-95FC-8A65D340689B}_is1') }
+        'Arm' { Throw 'Not supported.' }
+        'Arm64' { return $Insiders ? @('{69BD8F7B-65EB-4C6F-A14E-44CFA83712C0}_is1', '{0AEDB616-9614-463B-97D7-119DD86CCA64}_is1') : @('{D9E514E7-1A56-452D-9337-2990C0DC4310}_is1', '{A5270FC5-65AD-483E-AC30-2C276B63D0AC}_is1') }
+        Default { Throw 'Could not determine architecture.' }
+    }
+}
 
-        $insidersMachineInstallLocation = TryGetRegistryValue -Key "$($machineUninstallRegistry)\$($vsCodeInsidersMachineProductCode)" -Property $installLocationProperty
-        if ($insidersMachineInstallLocation) {
-            return $insidersMachineInstallLocation + $cmdPath
-        }
-    } else {
-        $cmdPath = 'bin\code.cmd'
-        $codeUserInstallLocation = TryGetRegistryValue -Key "$($userUninstallRegistry)\$($vsCodeUserProductCode)" -Property $installLocationProperty
-        if ($codeUserInstallLocation) {
-            return $codeUserInstallLocation + $cmdPath
+function Get-VSCodeCLIPath {
+    param (
+        [switch]$Insiders
+    )
+
+    # Get the available keys
+    $registryKeys = Get-OSArchitectureRegistryKey -Insiders:$Insiders.IsPresent
+    $registryHive = @('HKCU:\Software\Microsoft\Windows\CurrentVersion\Uninstall', 'HKLM:\Software\Microsoft\Windows\CurrentVersion\Uninstall', 'HKLM:\SOFTWARE\WOW6432Node\Microsoft\Windows\CurrentVersion\Uninstall')
+
+    if ($IsLinux) {
+        if ($Insiders) {
+            $InstallLocation = Join-Path ($env:PATH.Split([System.IO.Path]::PathSeparator) -match 'Microsoft VS Code Insiders') 'code-insiders'
+            if (Test-Path $InstallLocation -ErrorAction SilentlyContinue) {
+                return $InstallLocation
+
+            }
+        } else {
+            $InstallLocation = Join-Path ($env:PATH.Split([System.IO.Path]::PathSeparator) -match 'Microsoft VS Code') 'code'
+            if (Test-Path $InstallLocation -ErrorAction SilentlyContinue) {
+                return $InstallLocation
+            }
```

Comment:
You can move this to be outside of the if-else condition so that you don't have to repeat this code twice.